wa/commands/runstatus.py

```diff
@@ -0,0 +1,298 @@
+            else:
+                self._elapsed = self.ro.info.duration
+        return self._elapsed
+
+    @property
+    def job_outputs(self):
+        if self._job_outputs is None:
+            self._job_outputs = {
+                (j_o.id, j_o.label, j_o.iteration): j_o for j_o in self.ro.jobs
+            }
+        return self._job_outputs
+
+    @property
+    def projected_duration(self):
+        elapsed = self.elapsed_time.total_seconds()
+        proj = timedelta(seconds=elapsed * (len(self.jobs) / len(self.segmented['finished'])))
+        return proj - self.elapsed_time
+
+    def __init__(self, ro):
+        self.ro = ro
+        self._elapsed = None
+        self._p_duration = None
+        self._job_outputs = None
+        self._termwidth = None
+        self.formatter = _simple_formatter()
+        self.get_data()
+
+    def get_data(self):
+        self.jobs = [state for label_id, state in self.ro.state.jobs.items()]
+        rc = self.ro._combined_config.run_config
+        self.segmented = segment_jobs_by_state(self.jobs,
+                                                rc.max_retries,
+                                                rc.retry_on_status
+                                                )
+
+    def generate_run_header(self):
+        info = self.ro.info
+        
+        header = ('\n=========================\n'
+                  'Run Info'
+                  '\n=========================\n\n'
+                  )
+        header += "UUID: {}\n".format(info.uuid)
+        if info.run_name:
+            header += "Run name: {}\n".format(info.run_name)
+        if info.project:
+            header += "Project: {}\n".format(info.project)
+        if info.project_stage:
+            header += "Project stage: {}\n".format(info.project_stage)
+
+        if info.start_time:
+            duration = _seconds_as_smh(self.elapsed_time.total_seconds())
+            header += ("Start time: {}\n"
+                       "Duration: {:02}:{:02}:{:02}\n"
+                       ).format(info.start_time,
+                                duration[2], duration[1], duration[0],
+                                )
+            if len(self.segmented['finished']) and not info.end_time:
+                p_duration = _seconds_as_smh(self.projected_duration.total_seconds())
+                header += "Projected time remaining: {:02}:{:02}:{:02}\n".format(
+                    p_duration[2], p_duration[1], p_duration[0]
+                )
+
+            elif self.ro.info.end_time:
+                header += f"End time: {info.end_time}\n"
+
+        return header
+
+    def generate_job_summary(self):
+        total = len(self.jobs)
+        num_fin = len(self.segmented['finished'])
+
+        summary = ('\n=========================\n'
+                   'Job Summary'
+                   '\n=========================\n\n'
+                   )
+        summary += f'Total: {total}, Completed: {num_fin} ({(num_fin/total)*100}%)\n'
```

Comment:
This should check if the total number of runs is valid. In cases where the jobs were not generated correctly this results in a divide by 0 error.